File: backend/src/l1_support_bot/interface/dto/health.py
"""Health response contract."""

from typing import Literal

from pydantic import BaseModel, Field

ComponentStatus = Literal["available", "unavailable"]


class HealthResponse(BaseModel):
    status: Literal["healthy", "degraded"]
    version: str
    database: ComponentStatus
    vector_store: ComponentStatus
    llm: ComponentStatus
    embedding: ComponentStatus
    degraded_capabilities: list[str] = Field(default_factory=list)
    capabilities: dict[str, bool] = Field(default_factory=dict)

File: frontend/src/shared/api/health.ts
import { apiClient } from "./client";

export type HealthResponse = {
  status: "healthy" | "degraded";
  version: string;
  database: "available" | "unavailable";
  vector_store: "available" | "unavailable";
  llm: "available" | "unavailable";
  embedding: "available" | "unavailable";
  degraded_capabilities: string[];
  capabilities: Record<string, boolean>;
};

export async function getHealth(): Promise<HealthResponse> {
  return apiClient.request<HealthResponse>("/health");
}
